sic: add page (#5694)
tldr-pages/tldr: pages/linux/sic.md

# sic

> Simple IRC client.
> Part of the suckless tools.
> More information: <https://tools.suckless.org/sic/>.

- Connect to a host with the specified nickname:

> `sic -h {{irc.ofct.net}} -n {{$USER}}`

Reviewer: Maybe we could add the most basic example before this one?

```markdown
- Connect to the default host (irc.ofct.net) with the nickname set in the `$USER` variable:

`sic`
```

Then this one could be simply:

```markdown
- Connect to a host using a given nickname:

`sic -h {{host}} -n {{nickname}}`
```

Author: I like, if you submit the suggestion, i'll add it.

Reviewer: WDYT @bl-ue?

```suggestion
- Connect to the default host (irc.ofct.net) with the nickname set in the `$USER` variable:

`sic`

- Connect to a given host, using a given nickname:

`sic -h {{host}} -n {{nickname}}`
```

Reviewer 2: Looks good to me, one thing though: ~`$USER` variable~ → **`$USER` environment variable**, right, @waldyrious?

Reviewer: Right. I pondered that, and chose not to make the description too long, but it's best to be explicit. 👍

> - Connect to a host with the specified nickname:
> 
> `sic -h {{irc.ofct.net}} -n {{$USER}}`

Reviewer: WDYT @bl-ue?

```suggestion
- Connect to the default host (irc.ofct.net) with the nickname set in the `$USER` variable:

`sic`

- Connect to a given host, using a given nickname:

`sic -h {{host}} -n {{nickname}}`
```

> - Connect to a different host with PASS connection password:

Reviewer: I'm confused — what does the uppercase `PASS` refer to here?

Author: tbh, it's a copy paste from suckless docs; don't mind to remove the PASS part, turn it into irc password or something?

Reviewer: If my previous edit suggestion reaches consensus, then we could use matching language here as well, for consistency:

```suggestion
- Connect to a given host, using a given nickname and password:
```

Though before we make this change, I think we need to clarify what happens in the previous example (passing `-n` but not `-p`) — does sic expect the password to be in an environment variable too (say, `$PASS`)? Does it ask for it interactively?

Author: I don't use it like that normally, but just tested this out with `$PASS` set as an environment variable without `-k {{pass}}`. Does not work. So I'll commit your suggestion.

# sic

> simple irc client.

> > Part of suckless.org tools.

Reviewer 3: Perhaps this will read better?
```suggestion
> Part of the suckless.org tools.
```

Author: fair enough.

> More information: <https://tools.suckless.org/sic/>.

> - Connect to host (default: irc.oftc.net) with nickname (default: $USER):

Reviewer 3: You don't specify the host here to connect to, so I'm not sure the description here makes sense. What about this?
```suggestion
- Connect to irc.oftc.net with a given nickname (default: `$USER`):
```

Author: Devil is in the details, the explain part mentions host and the default.
Compromise: how about `connect to default (irc.ofct.net) irc host` ? 
Though I like my version better.

Reviewer 2: How about instead `Connect to host with a given nickname:` and instead of `nickname` in the example itself use `{{$USER}}` and show add `-h {{irc.ofct.net}}`?

Reviewer 2: See, this is what I'm thinking of: https://github.com/tldr-pages/tldr/pull/5694/files#r607929529

> - Connect to host (default: irc.oftc.net) with nickname (default: $USER):
> 
> `sic -n {{nickname}}`

Reviewer 2: I'm thinking this (from https://github.com/tldr-pages/tldr/pull/5694/files#r607888085)
```suggestion
- Connect to a host with the specified nickname:

`sic -h {{irc.ofct.net}} -n {{$USER}}`
```

Author: seems legit

- Connect to a different host with PASS connection password:

> `sic -h {{host}} -n {{nickname}} -k {{password}}`

Reviewer 3: Specifying a password on the command line like this is not very secure. Is there a way to prompt the user for a password, or read it from a file?

Author: @sbrl No. 
have a look [here](https://tools.suckless.org/sic/) =]

Reviewer 2: This is unfortunate but it doesn't matter much.  It's their fault if the command is bad, we're just documenting how to use it.

Author: Exactly.

- Usage with highlighted alerts on mention of your username:

> `sic | awk '/username/ {printf "\a"}1'`

Reviewer 3: This example seems to demonstrate the usage of the `awk` command, rather than the `sic` command. Generally in tldr pages we try to keep pages focused on the command they are about.

Author: No, this example demonstrates how you can highlight a message when your username is mentioned when using `sic`.

Reviewer 2: Yeah, I'm not sure if we should keep this example.

> - Join a channel (stdin):

Reviewer 3: Not sure what you mean by `(stdin)` here, or in future examples.

Author: `sic` works on `stdin` input. Please see [usage](https://tools.suckless.org/sic/usage/) and/or the man page of `sic`.

Reviewer 2: @Waples I see what you mean—this is a TUI and therefore when you start the tool it offers a prompt for you to enter commands etc.. So just omit `(stdin)` here and in the next examples.

See other documented TUIs for reference, e.g.:
- [`rainbowstream.md`](https://github.com/tldr-pages/tldr/blob/master/pages/common/rainbowstream.md)
- [`rtv.md`](https://github.com/tldr-pages/tldr/blob/master/pages/common/rtv.md)
- [`vim.md`](https://github.com/tldr-pages/tldr/blob/master/pages/common/vim.md)

Author: Thanks @bl-ue, i'll implement that.
